Clang failure because of const to non-const cast

Building with clang 3.6.2 fails with this error:

```
tests/CppUTestExt/MockSupport_cTestCFile.c:38:20: error: cast from 'const void *' to 'char *' drops const qualifier [-Werror,-Wcast-qual]
    return (char*) object;
                   ^
```
The error is caused by the `MockTypeValueToStringFunction_c` implementation, which requires a return value of `char*`.

```
static char* typeNameValueToString(const void* object)
{
    return (char*) object;
}
```

--- a/examples/AllTests/MockDocumentationTest.cpp
+++ b/examples/AllTests/MockDocumentationTest.cpp
@@ -179,9 +179,9 @@ static  int equalMethod(const void* object1, const void* object2)
     return object1 == object2;
 }
 
-static char* toStringMethod(const void*)
+static const char* toStringMethod(const void*)
 {
-    return (char*) "string";
+    return "string";
 }
 
 TEST(MockDocumentation, CInterface)

--- a/include/CppUTestExt/MockSupport_c.h
+++ b/include/CppUTestExt/MockSupport_c.h
@@ -108,7 +108,7 @@ struct SMockExpectedCall_c
 };
 
 typedef int (*MockTypeEqualFunction_c)(const void* object1, const void* object2);
-typedef char* (*MockTypeValueToStringFunction_c)(const void* object1);
+typedef const char* (*MockTypeValueToStringFunction_c)(const void* object1);
 typedef void (*MockTypeCopyFunction_c)(void* dst, const void* src);
 
 typedef struct SMockSupport_c MockSupport_c;

--- a/tests/CppUTestExt/MockSupport_cTest.cpp
+++ b/tests/CppUTestExt/MockSupport_cTest.cpp
@@ -58,9 +58,9 @@ extern "C"{
         return object1 == object2;
     }
 
-    static char* typeNameValueToString(const void* PUNUSED(object))
+    static const char* typeNameValueToString(const void* PUNUSED(object))
     {
-        return (char*) "valueToString";
+        return "valueToString";
     }
 
 }

--- a/tests/CppUTestExt/MockSupport_cTestCFile.c
+++ b/tests/CppUTestExt/MockSupport_cTestCFile.c
@@ -33,9 +33,9 @@ static int typeNameIsEqual(const void* object1, const void* object2)
     return object1 == object2;
 }
 
-static char* typeNameValueToString(const void* object)
+static const char* typeNameValueToString(const void* object)
 {
-    return (char*) object;
+    return (const char*) object;
 }
 
 void all_mock_support_c_calls(void)

